js/lightbeam.js

```diff
@@ -20,6 +24,54 @@ const lightbeam = {
     });
   },
 
+  // Called from init() (isFirstParty = undefined)
+  // and redraw() (isFirstParty = 1 or 0).
+  async updateVars(isFirstParty) {
+    const numSitesVisited = document.getElementById('num-sites-visited');
+    const numThirdParties = document.getElementById('num-third-parties');
+
+    // initialize dynamic vars from storage
+    if (!this.dataGatheredSince) {
+      this.dataGatheredSince = await this.getDataGatheredSince();
+      const dataGatheredSince = document.getElementById('data-gathered-since');
+      dataGatheredSince.textContent = this.dataGatheredSince;
+    }
+    if (isFirstParty === undefined) {
```

Comment:
I'm not a massive fan of this being undefined, however too tired to think of anything better right now.